File: Exercism Codes/C#/PrimeFactors.cs
public static class PrimeFactors
{
    public static long[] Factors(long number)
    {
        List<long> factors = [];
        long factor = 2;
        while (number > 1)
        {
            if (number % factor == 0)
            {
                factors.Add(factor);
                number /= factor;
            }
            else
            {
                factor++;
            }
        }

        return [..factors];
    }
}

File: Exercism Codes/JavaScript/prime-factors.js
export const primeFactors = (number) => {
  const result = [];
  let x = 2;
  while (number > 1) {
    if (number % x == 0) {
      result.push(x);
      number /= x;
    }
    else {
      x++;
    }
  }
  return result;
};
